File: functions/models/business.py
from typing import Optional

from pydantic import BaseModel, Field


class Business(BaseModel):
    uid: str  # Changed from 'id'
    name: str  # Added - was missing
    description: str
    industry: str
    logoFilepath: str
    plDocFilepath: str
    projectedRevenue: float
    projectedExpenses: float
    projectedProfit: float  # Added - was missing
    valuation: float
    totalSharesIssued: int  # Fixed typo from 'totalSharesIssues'
    sharePrice: float
    dividendPercentage: float  # Fixed typo from 'divdendPercentage'
    isApproved: bool
    address: str
    goal: float  # Changed to float to match Dart
    numInvestors: int
    amountRaised: int
    yearFounded: str  # Fixed from 'year_founded'

    # These might be in Firestore but not in Dart model - keep optional
    sharesAvailable: int
    ticker: str

File: lib/models/business.dart
import "package:freezed_annotation/freezed_annotation.dart";

part "business.freezed.dart";
part "business.g.dart";

@freezed
sealed class Business with _$Business {
  const factory Business({
    required String uid,
    required String name,
    required String description,
    required String industry,
    required String? logoFilepath,
    required String? plDocFilepath,
    required double projectedRevenue,
    required double projectedExpenses,
    required double projectedProfit,
    required double valuation,
    required int totalSharesIssued,
    required double sharePrice,
    required double dividendPercentage,
    required bool isApproved,
    required String address,
    required double goal,
    required int numInvestors,
    required int amountRaised,
    required String yearFounded,
    required int sharesAvailable,
    required String ticker,
  }) = _Business;

  factory Business.fromJson(Map<String, dynamic> json) =>
      _$BusinessFromJson(json);
}
